### PR TITLE
Add ability to load PDF from HTTP + specify file system from string (useful for config files)

### PR DESCRIPTION
**Pull request by stellasia**
# Description

> **Note**
>
> Please provide a description of the work completed in this PR below
>
>

## Type of Change
- [x] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High
>
>

Complexity:

## How Has This Been Tested?
- [ ] Unit tests
- [ ] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: src/neo4j_graphrag/experimental/components/pdf_loader.py [hunk 1]
```diff
@@ -85,16 +82,21 @@ def load_file(
...
-        filepath: Path,
+        filepath: Union[str, Path],
         metadata: Optional[Dict[str, str]] = None,
-        fs: Optional[AbstractFileSystem] = None,
+        fs: Optional[Union[AbstractFileSystem, str]] = None,
     ) -> PdfDocument:
-        fs = fs or LocalFileSystem()
+        if not isinstance(filepath, str):
+            filepath = str(filepath)
+        if isinstance(fs, str):
```
Comment (alexthomas93) [inline src/neo4j_graphrag/experimental/components/pdf_loader.py:91]: Why pass a str for the filesystem and initialise it here and not just pass the `fsspec.filesystem` to the method directly? We could remove the `Union` part of the type hint if we did. We could also set the default to `LocalFileSystem`

File: src/neo4j_graphrag/experimental/pipeline/config/object_config.py [hunk 2]
```diff
@@ -254,6 +254,10 @@ class ComponentConfig(ObjectConfig[Component]):
     DEFAULT_MODULE = "neo4j_graphrag.experimental.components"
     INTERFACE = Component
 
+    def get_run_params(self, resolved_data: dict[str, Any]) -> dict[str, Any]:
+        self._global_data = resolved_data
+        return self.resolve_params(self.run_params_)
+
 
 class ComponentType(RootModel):  # type: ignore[type-arg]
```
Comment (alexthomas93) [inline src/neo4j_graphrag/experimental/pipeline/config/object_config.py:262]: Not part of this PR but do we know why this type: ignore comment is here?

**Review by alexthomas93 — APPROVED** [review]
LGTM, just a few minor comments

File: src/neo4j_graphrag/experimental/components/pdf_loader.py (hunk 1 above)
Comment (stellasia) [reply src/neo4j_graphrag/experimental/components/pdf_loader.py:91]: The reason for the string is to make it work with config files. We could have a parser in the config parser to initialize a file system though, but I thought it was easier to just support str since this is allowed by `fsspec`.

Comment (alexthomas93) [reply src/neo4j_graphrag/experimental/components/pdf_loader.py:91]: Makes sense. I don't have a strong view on whether or not it should be here or in the config parser so let's just keep it here for now

File: src/neo4j_graphrag/experimental/pipeline/config/object_config.py (hunk 2 above)
Comment (stellasia) [reply src/neo4j_graphrag/experimental/pipeline/config/object_config.py:262]: Yes, this is because `pydantic.RootModel` can be used either like this:
```
class ComponentType(RootModel):  # type: ignore[type-arg]
    root: Union[Component, ComponentConfig]
```
 or with a simple declaration like this:
```
ComponentType = RootModel[Union[Component, ComponentConfig]]
```
We use the former to add other methods to the class, but mypy is complaining that the type annotation `[]` is missing for RootModel.

Comment (alexthomas93) [reply src/neo4j_graphrag/experimental/pipeline/config/object_config.py:262]: Hmm, that makes sense. Is there no way to point mypy to `root` in your example to resolve this issue? Feels a bit wrong to have this here. Maybe something we could look at in another PR in the future?

Comment (stellasia) [reply src/neo4j_graphrag/experimental/pipeline/config/object_config.py:262]: I couldn't find a workaround but I have to admit I've not spent a lot of time on it. Will create a card so that we can investigate further.
